src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -266,6 +288,22 @@ private void setJournalNodes(HashMap<String, String> journalNodes) {
     }
   }
 
+  private void setNameNodeTaskNames(HashMap<String, String> nameNodeTaskNames) {
+    try {
+      set(NAMENODE_TASKNAMES_KEY, nameNodeTaskNames);
+    } catch (Exception e) {
+      log.error("Error while setting namenodes in persistent state", e);
```

Comment:
You are correct. I think we need better error handling in persistent state. Propagating the exception up will be better, but that will be a refactor of the structure of PersistentState and probably should be another PR. Meanwhile, in most cases, if there is an error here, it will be caught elsewhere. The framework should be able to react.